kernel/src/expressions/scalars.rs

```diff
@@ -141,6 +141,24 @@ impl From<String> for Scalar {
 // TODO: add more From impls
 
 impl PrimitiveType {
+    /// Check if the given precision and scale are valid for a decimal type.
+    pub fn check_decimal(precision: u8, scale: u8) -> DeltaResult<()> {
+        require!(
+            precision > 0 && precision <= 38,
```

Comment:
tiny readability nit:
```suggestion
            0 < precision && precision <= 38,
```
(makes it easier to see this is a range)